cluster/local/integration_tests.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+set -e
+
+scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+projectdir="${scriptdir}/../.."
+
+# get the build environment variables from the special build.vars target in the main makefile
+eval $(make --no-print-directory -C ${projectdir} build.vars)
+BUILD_IMAGE="${BUILD_REGISTRY}/${PROJECT_NAME}-amd64"
+
+helm_tag="$(cat ${projectdir}/_output/version)"
+CROSSPLANE_IMAGE="${DOCKER_REGISTRY}/${PROJECT_NAME}:${helm_tag}"
+K8S_CLUSTER="${BUILD_REGISTRY}-INTTESTS"
+
+CROSSPLANE_NAMESPACE="crossplane-system"
+
+# cleanup on exit
+function cleanup {
+    export KUBECONFIG=
+    "${KIND}" delete cluster --name="${K8S_CLUSTER}"
+}
+trap cleanup EXIT
+
+# create cluster
+"${KIND}" create cluster --name="${K8S_CLUSTER}"
+export KUBECONFIG="$("${KIND}" get kubeconfig-path --name="${K8S_CLUSTER}")"
+
+# tag crossplane image and load it to kind cluster
+docker tag "${BUILD_IMAGE}" "${CROSSPLANE_IMAGE}"
+"${KIND}" load docker-image "${CROSSPLANE_IMAGE}" --name="${K8S_CLUSTER}"
+
+# install tiller
+"${KUBECTL}" apply -f "${scriptdir}/helm-rbac.yaml"
+"${HELM}" init --service-account tiller
+# waiting for deployment "tiller-deploy" rollout to finish
+"${KUBECTL}" -n kube-system rollout status deploy/tiller-deploy --timeout=2m
+
+# install crossplane
+echo "installing helm package(s) into \"$ns\" namespace"
+"${HELM}" install --name "${PROJECT_NAME}" --namespace "${CROSSPLANE_NAMESPACE}" "${projectdir}/cluster/charts/${PROJECT_NAME}" --set image.pullPolicy=Never,imagePullSecrets=''
+
+# waiting for deployment "crossplane" rollout to finish
+"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" rollout status "deploy/${PROJECT_NAME}" --timeout=2m
+
+# ----------- integration tests
+# get the pods statuses
+podsstatuses=`"${KUBECTL}" -n "${CROSSPLANE_NAMESPACE}" get pods -o=jsonpath='{range .items[*]}{@.metadata.name}{":"}{@.status.phase}{" "}'`
+
+# check for minimum number of pods created
+MIN_CROSSPLANE_PODS=2
+IFS=' ' read -ra PODS_ARRAY <<< "$podsstatuses"
+PODS_COUNT=${#PODS_ARRAY[@]}
+if (( ${PODS_COUNT} < ${MIN_CROSSPLANE_PODS} )); then
+    echo "number of created pods are ${PODS_COUNT}, which is less than the minimum of ${MIN_CROSSPLANE_PODS}"
+    exit -1
+fi
+
+# check for all pods to be running
+for pod_status in ${podsstatuses}
+do
+    IFS=':' read -ra parts <<< "$pod_status"
+    echo "${parts[0]} is ${parts[1]}"
+    if [ ${parts[1]} != "Running" ]; then
+        echo "integration test failed"
+        exit -1
+    fi
+done
```

Comment:
Maybe I am missing it, where are the clean-up steps?

Reply:
line 22. `trap` is executed when exiting